pkg/internal/netolly/ebpf/record.go

```diff
@@ -73,6 +73,8 @@ func (fm *NetFlowMetrics) Accumulate(src *NetFlowMetrics) {
 	// time == 0 if the value has not been yet set
 	if fm.StartMonoTimeNs == 0 || fm.StartMonoTimeNs > src.StartMonoTimeNs {
 		fm.StartMonoTimeNs = src.StartMonoTimeNs
+		// set Direction here, because the correct value is in the first packet only (socket_filter case)
```

Comment:
Very cool idea!

Reply:
thanks for pointing it. Now it is not socket_filter case only 😉 . Changed comment.